### DISCUSSION TITLE
Hook_queue never called

### DISCUSSION THREAD

**Discussion by florentjb** [haraka/Haraka#3283] [Q&A]
Hello, 

I try to put emails in a rabbitmq queue with the plugin which works well
I don't understand why hook_queue is never called ( last version 3.x) 
I'm trying to use the rabbitmq plugin in a fairly basic way as presented in the documentation:
Here is the debug:

**plugin/config** 
tls
auth_redis       //  [ this is my plugin to auth smtp user ] 
queue/rabbitmq   

**Start haraka :**  
 /
loaded TLD files:
  1=1455
  2=9821
  3=5419
loaded 9566 Public Suffixes
loglevel: DATA
log format: DEFAULT
[WARN] [-] [core] smtp.ini.nodes unset, using 1, see https://github.com/haraka/Haraka/wiki/Performance-Tuning
Starting up Haraka version 3.0.3
[INFO] [-] [core] Loading plugins
[INFO] [-] [core] Loading plugin: tls
[DEBUG] [-] [core] no timeout in tls.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin tls timeout is: 30s
[INFO] [-] [core] loading tls.ini
[DEBUG] [-] [core] loading dhparams from dhparams.pem
[DEBUG] [-] [core] registered hook capabilities to tls.advertise_starttls priority 0
[DEBUG] [-] [core] registered hook unrecognized_command to tls.upgrade_connection priority 0
[DEBUG] [-] [core] registered hook disconnect to tls.hook_disconnect priority 0
[INFO] [-] [core] Loading plugin: auth_redis
[DEBUG] [-] [core] no timeout in auth_redis.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin auth_redis timeout is: 30s
[DEBUG] [-] [core] no timeout in auth/auth_base.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin auth/auth_base timeout is: 30s
[DEBUG] [-] [core] registered hook capabilities to auth_redis.hook_capabilities priority 0
[DEBUG] [-] [core] registered hook unrecognized_command to auth_redis.hook_unrecognized_command priority 0
[INFO] [-] [core] Loading plugin: queue/rabbitmq
[DEBUG] [-] [core] no timeout in queue/rabbitmq.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin queue/rabbitmq timeout is: 30s
[DEBUG] [-] [core] About to connect and initialize queue object
[DEBUG] [-] [core] About to Create connection with server
[DEBUG] [-] [core] Finished initiating : undefined
[DEBUG] [-] [core] registered hook queue to queue/rabbitmq.hook_queue priority 0
[DEBUG] [-] [server] running init_master hooks
[DEBUG] [-] [core] Connection is ready, will try making exchange
[NOTICE] [-] [core] worker started worker=1 pid=670556
loaded TLD files:
  1=1455
  2=9821
  3=5419
loaded 9566 Public Suffixes
loglevel: DATA
log format: DEFAULT
[WARN] [-] [core] smtp.ini.nodes unset, using 1, see https://github.com/haraka/Haraka/wiki/Performance-Tuning
Starting up Haraka version 3.0.3
[INFO] [-] [core] Loading plugins
[INFO] [-] [core] Loading plugin: tls
[DEBUG] [-] [core] no timeout in tls.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin tls timeout is: 30s
[INFO] [-] [core] loading tls.ini
[DEBUG] [-] [core] loading dhparams from dhparams.pem
[DEBUG] [-] [core] registered hook capabilities to tls.advertise_starttls priority 0
[DEBUG] [-] [core] registered hook unrecognized_command to tls.upgrade_connection priority 0
[DEBUG] [-] [core] registered hook disconnect to tls.hook_disconnect priority 0
[INFO] [-] [core] Loading plugin: auth_redis
[DEBUG] [-] [core] no timeout in auth_redis.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin auth_redis timeout is: 30s
[DEBUG] [-] [core] no timeout in auth/auth_base.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin auth/auth_base timeout is: 30s
[DEBUG] [-] [core] registered hook capabilities to auth_redis.hook_capabilities priority 0
[DEBUG] [-] [core] registered hook unrecognized_command to auth_redis.hook_unrecognized_command priority 0
[INFO] [-] [core] Loading plugin: queue/rabbitmq
[DEBUG] [-] [core] no timeout in queue/rabbitmq.timeout
[DEBUG] [-] [core] no timeout in plugin_timeout
[DEBUG] [-] [core] plugin queue/rabbitmq timeout is: 30s
[DEBUG] [-] [core] About to connect and initialize queue object
[DEBUG] [-] [core] About to Create connection with server
[DEBUG] [-] [core] Finished initiating : undefined
[DEBUG] [-] [core] registered hook queue to queue/rabbitmq.hook_queue priority 0
[NOTICE] [-] [core] Listening on 1[xxx.xxx.xxx.xxx:587](xxx.xxx.xxx.xxx:587)
[DEBUG] [-] [server] running init_child hooks
[NOTICE] [-] [core] worker 1 listening on xxx.xxx.xxx.xxx:587
[DEBUG] [-] [core] Connection is ready, will try making exchange

**when i send mail with swakes  :**     
  swaks -t n******@gmail.com -s xxx.xxxx.xxxx.xxx587 -tls -au user -ap passwd  -f toto@mta1.********. com  

     ....

<~  250 Message Queued (63E39DE2-8C52-45B5-B871-FCB08498BFA7.1)
 ~> QUIT
 
 (THAT OK ! ) 

**in haraka** 

[DEBUG] [-] [core] addOCSP: not available
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] connect ip=xx.xxx.xxx.xxx port=57122 local_ip=xx.xxx.xxx.xxx local_port=587
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running connect_init hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running connect_init_respond
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running lookup_rdns hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running connect hooks
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 220 netitm ESMTP Haraka/3.0.3 ready
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: EHLO netitm.local state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running ehlo hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=capabilities plugin=tls function=advertise_starttls params="" retval=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hook in auth_redis plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=capabilities plugin=auth_redis function=hook_capabilities params="" retval=CONT msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-netitm Hello ipxxxxxxxxx.eu [xx.xxx.xxx.xxx] Haraka is at your service.
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-PIPELINING
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-8BITMIME
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-SMTPUTF8
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-SIZE 0
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250 STARTTLS
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: STARTTLS state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in tls plugin
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 220 Go ahead.
[DEBUG] [-] [core] Upgrading to TLS
[DEBUG] [-] [core] TLS secured.
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [tls] secured: cipher=TLS_AES_256_GCM_SHA384 version=TLSv1.3 verified=false
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params=STARTTLS retval=OK msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: EHLO netitm .local state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running ehlo hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=capabilities plugin=tls function=advertise_starttls params="" retval=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running capabilities hook in auth_redis plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=capabilities plugin=auth_redis function=hook_capabilities params="" retval=CONT msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-netitm  Hello ipxxxxxxx.eu [xx.xxx.xxx.xxx]Haraka is at your service.
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-PIPELINING
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-8BITMIME
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-SMTPUTF8
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250-SIZE 0
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 250 AUTH PLAIN LOGIN
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: AUTH LOGIN state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params=AUTH retval=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in auth_redis plugin
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 334 VXNlcm5hbWU6
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=auth_redis function=hook_unrecognized_command params=AUTH retval=OK msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: dGF0YQ== state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params="dGF0YQ==" retval=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in auth_redis plugin
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 334 UGFzc3dvcmQ6
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=auth_redis function=hook_unrecognized_command params="dGF0YQ==" retval=OK msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: dGF0YQ== state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=tls function=upgrade_connection params="dGF0YQ==" retval=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] running unrecognized_command hook in auth_redis plugin
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] S: 235 2.7.0 Authentication successful
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core]  hook=unrecognized_command plugin=auth_redis function=hook_unrecognized_command params="dGF0YQ==" retval=OK msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7] [core] C: MAIL FROM:<toto@mta1.xx.xxx.xxx.xxx> state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running mail hooks
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] sender <toto@mta1.xx.xxx.xxx.xxx.com> code=CONT msg=""
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] S: 250 sender <toto@mta1.xx.xxx.xxx.xxx.com> OK
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: RCPT TO:<n********@gmail.com> state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running rcpt hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running rcpt_ok hooks
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] recipient <netitbee@gmail.com> code=OK msg="" sender=toto@mta1.xx.xxx.xxx.xxx.com
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] S: 250 recipient <netitbee@gmail.com> OK
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: DATA state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running data hooks
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] S: 354 go ahead, make my day
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: Date: Fri, 08 Mar 2024 09:52:31 +0000\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: To: netitbee@gmail.com\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: From: toto@mta1.xx.xxx.xxx.xxx.com\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: Subject: test Fri, 08 Mar 2024 09:52:31 +0000\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: Message-Id: <20240308095231.671194@xx.xxx.xxx.xxx.local>\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: X-Mailer: swaks v20201014.0 jetmore.org/john/code/swaks/\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: \r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: This is a test mailing\r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: \r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: \r
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: .\r
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running data_post hooks
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] message mid=<20240308095231.671194@xx.xxx.xxx.xxx.local> size=287 rcpts=1/0/0 delay=0 code=CONT msg=""
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running queue_outbound hooks
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] queue code=CONT msg="Message Queued (63E39DE2-8C52-45B5-B871-FCB08498BFA7.1)"
[DEBUG] [-] [core] [outbound] Sending email as a transaction
[DEBUG] [-] [core] running pre_send_trans_email hooks
[INFO] [-] [core] [outbound] Transaction delivery for domain: gmail.com
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running queue_ok hooks
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] queue code=OK msg="Message Queued (63E39DE2-8C52-45B5-B871-FCB08498BFA7.1)"
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] S: 250 Message Queued (63E39DE2-8C52-45B5-B871-FCB08498BFA7.1)
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running reset_transaction hooks
[INFO] [-] [core] loading tls.ini
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] C: QUIT state=1
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running quit hooks
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] S: 221 xx.xxx.xxx.xxx closing connection. Have a jolly good day.
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] client has disconnected
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running disconnect hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] client has disconnected
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] running disconnect hook in tls plugin
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] client has disconnected
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core]  hook=disconnect plugin=tls function=hook_disconnect params="" retval=CONT msg=""
[NOTICE] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1] [core] disconnect ip=xx.xxx.xxx.xxx  rdns=ipxx.xxx.xxx.xxx.eu helo=netitm**.local relay=Y early=N esmtp=Y tls=Y pipe=N errors=0 txns=1 rcpts=1/0/0 msgs=1/0/0 bytes=287 lr="" time=0.046
[DEBUG] [-] [outbound] todo header length: 498
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] running send_email hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] Sending mail: 1709891551544_1709891551544_0_671131_RoA6tu_1_netitm
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] running get_mx hooks
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] DNS (A) for gmail-smtp-in.l.google.com -> 142.251.168.26
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] delivering from: netitr to: 142.251.168.26:25 (0) (0)
[DEBUG] [7545B2B8-8F9B-47E7-9C0C-0DFBA6FDDCB6] [core] [outbound] created. host: 142.251.168.26 port: 25
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 220 mx.google.com ESMTP r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: EHLO netitm
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-mx.google.com at your service, [xx.xxx.xxx.xxx]\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-SIZE 157286400\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-8BITMIME\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-STARTTLS\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-ENHANCEDSTATUSCODES\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-PIPELINING\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-CHUNKING\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250 SMTPUTF8\r\n
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] Trying TLS for domain: gmail.com, host: 142.251.168.26
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: STARTTLS
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 220 2.0.0 Ready to start TLS\r\n
[DEBUG] [-] [core] client TLS upgrade in progress, awaiting secured.
[DEBUG] [-] [core] client TLS secured.
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] secured verified=true cipher=TLS_AES_256_GCM_SHA384 version=TLSv1.3 cn=mx.google.com organization="" issuer="Google Trust Services LLC" expires="Apr 29 08:19:07 2024 GMT" fingerprint=B4:CA:F9:B8:29:C9:2C:F4:3E:4E:AB:0E:17:C1:2D:33:C9:D9:22:83
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: EHLO netitm
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-mx.google.com at your service, [xx.xxx.xxx.xxx]\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-SIZE 157286400\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-8BITMIME\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-ENHANCEDSTATUSCODES\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-PIPELINING\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250-CHUNKING\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250 SMTPUTF8\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: MAIL FROM:<toto@mta1.********.com>
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250 2.1.0 OK r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: RCPT TO:<n********@gmail.com>
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 250 2.1.5 OK r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: DATA
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 354  Go ahead r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp\r\n
[DATA] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: Received: (Haraka outbound); Fri, 08 Mar 2024 09:52:31 +0000\r
Authentication-Results: netitmr;\r
	auth=pass (login)\r
Received: from netitmailer.local (ipxx.xxx.xxx.xxx.eu [xx.xxx.xxx.xxx])\r
	by netitm (Haraka/3.0.3) with ESMTPSA id 63E39DE2-8C52-45B5-B871-FCB08498BFA7.1\r
	envelope-from <toto@mta1.xx.xxx.xxx.xxx.com>\r
	tls TLS_AES_256_GCM_SHA384 (authenticated bits=0);\r
	Fri, 08 Mar 2024 09:52:31 +0000\r
Date: Fri, 08 Mar 2024 09:52:31 +0000\r
To: netitbee@gmail.com\r
From: toto@mta1.*******.com\r
Subject: test Fri, 08 Mar 2024 09:52:31 +0000\r
Message-Id: <20240308095231.671194@*******.local>\r
X-Mailer: swaks v20201014.0 jetmore.org/john/code/swaks/\r
\r
This is a test mailing\r
\r
\r
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: .
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26 This mail has been blocked because the sender is unauthenticated.\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26 Gmail requires all senders to authenticate with either SPF or DKIM.\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26 \r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26  Authentication results:\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26  DKIM = did not pass\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26  SPF [mta1.******.com] with ip: [xx.xxx.xxx.xxx] = did not pass\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26 \r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550-5.7.26  For instructions on setting up authentication, go to\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] S: 550 5.7.26  https://support.google.com/mail/answer/81126#authentication r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp\r\n
[PROTOCOL] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] C: QUIT
[INFO] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] bouncing mail: 550 5.7.26 This mail has been blocked because the sender is unauthenticated. 5.7.26 Gmail requires all senders to authenticate with either SPF or DKIM. 5.7.26  5.7.26  Authentication results: 5.7.26  DKIM = did not pass 5.7.26  SPF [mta1.******.com] with ip: [xx.xxx.xxx.xxx] = did not pass 5.7.26  5.7.26  For instructions on setting up authentication, go to 5.7.26  https://support.google.com/mail/answer/81126#authentication r12-20020a056000014c00b0033e7780c401si460136wrx.768 - gsmtp
[DEBUG] [63E39DE2-8C52-45B5-B871-FCB08498BFA7.1.1] [outbound] running bounce hooks
[DEBUG] [-] [core] [outbound] release_client: 7545B2B8-8F9B-47E7-9C0C-0DFBA6FDDCB6 142.251.168.26:25 to undefined
[INFO] [-] [core] [outbound] Sending email via params
[INFO] [-] [core] [outbound] Created transaction: 76C4E767-19E3-4EFC-B337-578FEAE95EAD
[INFO] [-] [core] [outbound] Adding missing Message-Id header
[INFO] [-] [core] [outbound] Adding missing Date header
[DEBUG] [-] [core] adding results store
[DEBUG] [-] [core] running pre_send_trans_email hooks
[INFO] [-] [core] [outbound] Transaction delivery for domain: mta1.*******.com
[DEBUG] [-] [outbound] todo header length: 404
[DEBUG] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] running send_email hooks
[DEBUG] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] Sending mail: 1709891552263_1709891552263_0_671131_BFFti5_2_*****
[DEBUG] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] running get_mx hooks
[DEBUG] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] delivering from: ****** to: xx.xxx.xxx.xxx:25 (0) (0)
[DEBUG] [BC4988B6-7C3A-40F4-9668-EA36D665C3CC] [core] [outbound] created. host: 1xx.xxx.xxx.xxx port: 25
[NOTICE] [-] [core] [outbound] Failed to get socket: Outbound connection error: Error: connect ECONNREFUSED xx.xxx.xxx.xxx:25
[DEBUG] [-] [core] Temp fail for: Tried all MXs
[DEBUG] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] running deferred hooks
[INFO] [76C4E767-19E3-4EFC-B337-578FEAE95EAD.1] [outbound] Temp failing 1709891552263_1709891552263_0_671131_BFFti5_2_netitm  for 64 seconds: Tried all MXs

I anonymized the IP and the domain obviously

Thanks to anyone who can help me out

**Comment by puth** [comment]
You need to call hook_rcpt first:
```
exports.hook_rcpt = function (next, connection, params) {
    next(OK);
}
```
